pkg/controller/common/helpers.go

```diff
@@ -250,30 +252,145 @@ func IgnParseWrapper(rawIgn []byte) (interface{}, error) {
 }
 
 // ParseAndConvertConfig parses rawIgn for both V2 and V3 ignition configs and returns
-// a V2 or an error.
-func ParseAndConvertConfig(rawIgn []byte) (ign2types.Config, error) {
+// a V3 or an error.
+func ParseAndConvertConfig(rawIgn []byte) (ign3types.Config, error) {
 	ignconfigi, err := IgnParseWrapper(rawIgn)
 	if err != nil {
-		return ign2types.Config{}, errors.Wrapf(err, "failed to parse Ignition config")
+		return ign3types.Config{}, errors.Wrapf(err, "failed to parse Ignition config")
 	}
 
 	switch typedConfig := ignconfigi.(type) {
+	case ign3types.Config:
+		return ignconfigi.(ign3types.Config), nil
 	case ign2types.Config:
-		return ignconfigi.(ign2types.Config), nil
+		convertedIgnV3, err := ConvertIgnition2to3(ignconfigi.(ign2types.Config))
+		if err != nil {
+			return ign3types.Config{}, errors.Wrapf(err, "failed to convert Ignition config spec v2 to v3")
+		}
+		return convertedIgnV3, nil
+	default:
+		return ign3types.Config{}, errors.Errorf("unexpected type for ignition config: %v", typedConfig)
+	}
+}
+
+// Test: have some situations de-duplicate and see what happens, will merge into main function
+func ParseAndConvertConfigWithDeduplication(rawIgn []byte) (ign3types.Config, error) {
```

Comment:
will probably merge this into the main function, just trying to think if there's scenarios we won't want to de-duplicate